### PR TITLE
compilerArgs entries fix

### PR DESCRIPTION
**Pull request by andreeis**
Fix for bug https://github.com/microsoft/vscode-makefile-tools/issues/142. 
Send different compilerArgs entries for bundled switches or pairs of switch-value if separated by space as a shell would see them.

### CODE REVIEW

File: src/parser.ts [hunk 1]
```diff
@@ -414,30 +425,74 @@ function parseAnySwitchFromToolArguments(args: string, excludeArgs: string[]): s
...
+    if (!util.checkFileExistsSync(parseCompilerArgsScriptFile)) {
+        util.writeFile(parseCompilerArgsScriptFile, parseCompilerArgsScriptContent);
+    }
+
+    let scriptArgs: string[] = [];
+    let runCommand: string;
+    if (process.platform === 'win32') {
+        runCommand = "cmd";
+        scriptArgs.push("/c");
+        scriptArgs.push(parseCompilerArgsScriptFile);
```
Comment (bobbrow) [inline src/parser.ts:465]: Should we surround the file name with double quotes or do you guarantee no spaces in the file name?

Comment (bobbrow) [reply src/parser.ts:465]: similar question for the *nix invocation - might need single quotes? (I forget the rules there)

Comment (bobbrow) [reply src/parser.ts:465]: ok, so the way this is called, you might not need it on windows.  But I still have a concern about linux.

File: src/parser.ts [hunk 2]
```diff
@@ -414,30 +425,74 @@ function parseAnySwitchFromToolArguments(args: string, excludeArgs: string[]): s
...
-        }
+        };
 
-        match1 = match2;
+        let stderr: any = (result: string): void => {
+            logger.messageNoCR(`Error while running the compiler args parser script '${parseCompilerArgsScriptFile}'` +
+                               `for regions "${compilerArgRegions}": "${result}"`, "Normal");
+        };
+
+        const result: util.SpawnProcessResult = await util.spawnChildProcess(runCommand, scriptArgs, vscode.workspace.rootPath || "", stdout, stderr);
```
Comment (bobbrow) [inline src/parser.ts:491]: `vscode.workspace.rootPath` is deprecated and should not be used.  We'll need to get this value out of `vscode.workspace.workspaceFolders`

File: src/parser.ts (hunk 1 above)
Comment (andreeis) [reply src/parser.ts:465]: Yes a file with spaces was not able to work. I fixed on linux with single quotes around file name but on windows it's broken too, but adding double quotes around file name is not helping. Investigating how to fix on windows too.
Luckily, the extension is not generating name with space but if TEMP happens to be defined to a path with space then we would have this problem. Thanks for thinking of this scenario.

Comment (andreeis) [reply src/parser.ts]: I fixed windows as well (I was sending parameters of script as parameters of cmd.exe and confused it).
I tested these diffs again on both.

File: src/parser.ts (hunk 2 above)
Comment (andreeis) [reply src/parser.ts]: Right, done.

**Review by bobbrow — APPROVED** [review]

**Pull request merged**